pkgs/native_assets_builder/lib/src/build_runner/build_runner.dart

```diff
@@ -154,10 +163,12 @@ class NativeAssetsBuildRunner {
       // here whether the config is equal.
 
       setMetadata(config.target, packageName, buildOutput?.metadata);
-      return buildOutput!.assets;
+      _assets.addAll(buildOutput!.assets);
+      _dependencies.addAll(buildOutput.dependencies.dependencies);
+      return;
```

Comment:
Nice catch, I could come up with at least one scenario in which this can be problematic: https://github.com/dart-lang/native/issues/111. Will address in a separate PR.